src/types.rs

```diff
@@ -244,6 +244,28 @@ pub struct NewAccount<'a> {
     ///
     /// Setting this to `false` has not been tested.
     pub only_return_existing: bool,
+    /// External account binding
+    pub external_account_binding: Option<ExternalAccountBinding<'a>>,
+}
+
+/// External account binding as described in RFC 8555 (section 7.3.4)
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExternalAccountBinding<'a> {
+    /// EAB Key ID
+    pub key_id: &'a str,
+    /// EAB HMAC key
+    pub hmac_key: &'a str,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct JoseNewAccount<'a> {
+    pub(crate) contact: &'a [&'a str],
+    pub(crate) terms_of_service_agreed: bool,
+    pub(crate) only_return_existing: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) external_account_binding: Option<JoseJson>,
```

Comment:
I don't really like the duplication of the `NewAccount` fields here. Do you think we could do this by creating a custom `Serialize` implementation for `ExternalAccountBinding` instead?

Reply:
I think it's not that simple because we can't pass the necessary parameters for signing `ExternalAccountBinding` to the custom `Serialize` implementation.

Reply:
Here is another possible solution:

```
#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct NewAccount<'a> {
    pub contact: &'a [&'a str],
    pub terms_of_service_agreed: bool,
    pub only_return_existing: bool,
    #[serde(skip)]
    pub external_account_binding: Option<ExternalAccountBinding<'a>>,
}

#[derive(Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct NewAccountWithSignedEab<'a> {
    #[serde(flatten)]
    pub(crate) new_account: NewAccount<'a>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub(crate) external_account_binding: Option<JoseJson>,
}
```